ssl/ssl_lib.c

```diff
@@ -2837,13 +2852,24 @@ void SSL_CTX_set_npn_select_cb(SSL_CTX *ctx,
 int SSL_CTX_set_alpn_protos(SSL_CTX *ctx, const unsigned char *protos,
                             unsigned int protos_len)
 {
-    OPENSSL_free(ctx->ext.alpn);
-    ctx->ext.alpn = OPENSSL_memdup(protos, protos_len);
-    if (ctx->ext.alpn == NULL) {
+    unsigned char *alpn;
+
+    if (protos_len == 0 || protos == NULL) {
+        OPENSSL_free(ctx->ext.alpn);
+        ctx->ext.alpn = NULL;
         ctx->ext.alpn_len = 0;
```

Comment:
Why are you freeing existing values only in the event that they are clearing the ALPN values? Won't this lead to a potential memory leak if they set it to some non-NULL value, and then set it again to some other non-NULL value?

Reply:
Damn, looks as tough I missed the free for the set below :/
I rewrote this with the check function, and that's when it disappeared.

Reply:
Considering the unit tests set the ALPN value multiple times on the same resource, I'm very surprised that the sanitizer CI didn't catch this!